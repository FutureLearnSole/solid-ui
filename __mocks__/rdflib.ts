export const Util =  {
  mediaTypeClass: sym
}

export function sym () {
  return {
    dir: sym,
    uri: 'uri',
    value: '',
    doc: () => {
      equals: () => {}
    },
    sameTerm: () => false
  }  
}  

export const uri = {
  join: () => {}
}
export function st() {

}
export function graph() {
  return {
    any: () => {
      return sym()
    },
    each: () => {
      return []
    },
    query: () => {
      return []
    },
    fetcher: {
      load: () => {},
      nowOrWhenFetched: () => Promise.resolve()
    },
    findTypeURIs: () => {
      return []
    },
    findSuperClassesNT: () => {
      return []
    },
    statementsMatching: () => {
      return []
    },
    sym
  }
}
export function fetcher() {
  return {}
}
export class Fetcher {
  requested: any
  constructor() {
    this.requested = {}
  }
  load () {
  }
}
export function namedNode() {
  return {}
}
export function NamedNode() {
}
export function Namespace() {
  return () => {}
}
export class UpdateManager {
}
