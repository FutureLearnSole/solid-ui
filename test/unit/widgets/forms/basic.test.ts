import { namedNode } from 'rdflib'
import {
  fieldLabel,
  fieldStore,
  basicField
} from '../../../../src/widgets/forms/basic'

describe('fieldLabel', () => {
  it('exists', () => {
    expect(fieldLabel).toBeInstanceOf(Object)
  })
  it('runs', () => {
    // expect(fieldLabel(document, namedNode('http://example.com/#this'), null)).toBeInstanceOf(
    //   HTMLAnchorElement
    // )
  })
  it.skip(' ...', () => {
    // expect(fieldLabel('document', undefined, 'form').toBe())
  })
})

describe('fieldStore', () => {
  it('exists', () => {
    expect(fieldStore).toBeInstanceOf(Object)
  })
  it.skip('runs', () => {
    // expect(fieldStore(null, null, null)).toEqual({
    //   termType: 'NamedNode',
    //   value: 'http://example.com/'
    // })
  })
  it.skip('returns def when there is no matching statement', () => {
    // const statementMatching = jest.fn()
    // statementMatching.mockReturnValueOnce(null)
    // expect(fieldStore('subject', 'predicate', 'def')).toBe('def')
  })
})

describe('basicField', () => {
  it('exists', () => {
    expect(fieldStore).toBeInstanceOf(Function)
  })
  it('runs', () => {
    const container = document.createElement('div')
    const already = {}
    const subject = namedNode('http://example.com/#this')
    const form = namedNode('http://example.com/#form')
    const store = namedNode('http://example.com/#store')
    const callbackFunction = jest.fn() // TODO: https://github.com/solid/solid-ui/issues/263
    expect(
      basicField(
        document,
        container,
        already,
        subject,
        form,
        store,
        callbackFunction
      )
    ).toMatchSnapshot()
  })
})
