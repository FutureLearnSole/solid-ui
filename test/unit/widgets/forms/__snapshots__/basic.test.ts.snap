// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basicField runs 1`] = `
<tr>
  <td
    class="formFieldName"
    style="  vertical-align: middle;"
  />
  <td
    class="formFieldValue"
  />
  Error: No property given for text field: &lt;http://example.com/#form&gt;
</tr>
`;
